hbase/Dockerfile

```diff
@@ -1,45 +1,63 @@
 # syntax=docker/dockerfile:1
-FROM stackable/image/java-base
+FROM stackable/image/java-base AS builder
 
 ARG PRODUCT
 ARG PHOENIX
-ARG RELEASE
-
-LABEL name="Apache HBase" \
-      maintainer="info@stackable.tech" \
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Apache HBase." \
-      description="This image is deployed by the Stackable Operator for Apache HBase."
+ARG JAVA
+ARG JMX_EXPORTER
+ARG HADOOP_M2
 
 # https://github.com/hadolint/hadolint/wiki/DL4006
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
+RUN echo "${JAVA}"
+RUN echo "${PRODUCT}"
+
+# zip required for the log4shell stuff
 RUN microdnf update && \
-    microdnf install tar gzip zip && \
-    microdnf install python3 python3-pip && \
+    microdnf install \
+    git \
+    gzip \
+    java-${JAVA}-openjdk-devel \
+    maven \
+    tar \
+    zip && \
     microdnf clean all
 
-RUN ln -s /usr/bin/python3 /usr/bin/python && \
-    ln -s /usr/bin/pip3 /usr/bin/pip
-
-ENV HOME=/stackable
+# The patch was generated by calling https://github.com/apache/hbase/pull/5233/commits/4f22587d02311b91c8371d5e6b9a2758d39ab26e.patch
 
 COPY --chown=stackable:stackable hbase/stackable /stackable
 COPY hbase/licenses /licenses
 
 USER stackable
 WORKDIR /stackable
 
-RUN curl -L https://repo.stackable.tech/repository/packages/hbase/hbase-${PRODUCT}-bin.tar.gz | tar -xzC . && \
-    ln -s /stackable/hbase-${PRODUCT} /stackable/hbase && \
-    curl https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-0.16.1.jar \
-    -o /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar && \
-    chmod -x /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar && \
-    curl -L https://repo.stackable.tech/repository/packages/phoenix/phoenix-hbase-${PHOENIX}-bin.tar.gz | tar -xzC . && \
-    ln -s /stackable/phoenix-hbase-${PHOENIX}-bin /stackable/phoenix && \
-    ln -s /stackable/phoenix/phoenix-server-hbase-${PHOENIX}.jar /stackable/hbase/lib/phoenix-server-hbase-${PHOENIX}.jar
+# The release scripts of HBase also run the build twice (three times in fact, once again to build the site which we skip here). I chose to replicate that exact behavior for consistency so please don't merge the two mvn runs into one unless you really know what you're doing!
+RUN curl --fail -L "https://repo.stackable.tech/repository/packages/hbase/hbase-${PRODUCT}-src.tar.gz" | tar -xzC .
+RUN mv hbase-${PRODUCT} hbase-${PRODUCT}-src
+RUN chmod +x patches/apply_patches.sh
+RUN patches/apply_patches.sh ${PRODUCT}
+RUN cd /stackable/hbase-${PRODUCT}-src/ && \
```

Comment:
Could be summarized in to one run command (to make hadolint happy). But since its the builder i do not really have a strong opinion.

Reply:
I find this easier to read and it doesn't matter for the builder
But I have no strong preference either way.